pkg/kubectl/cmd/cmd_test.go

```diff
@@ -73,6 +73,12 @@ func TestKubectlCommandHandlesPlugins(t *testing.T) {
 			expectPlugin:     "plugin/testdata/kubectl-foo",
 			expectPluginArgs: []string{"--bar"},
 		},
+		{
+			name:             "test that flags before plugin name are passed to plugin",
+			args:             []string{"kubectl", "--flag1=bar", "-n", "baz", "foo", "--flag2", "-f3", "qux", "arg"},
```

Comment:
This does assume that every flag has an argument to it, which is consistent with what we have in kubectl.